packages/flutter/lib/src/foundation/collections.dart

```diff
@@ -117,3 +117,215 @@ int binarySearch<T extends Comparable<Object>>(List<T> sortedList, T value) {
   }
   return -1;
 }
+
+/// Limit below which merge sort defaults to insertion sort.
+const int _MERGE_SORT_LIMIT = 32;
+
+/// Sorts a list between [start] (inclusive) and [end] (exclusive) using the
+/// merge sort algorithm.
+///
+/// If [compare] is omitted, this defaults to calling [Comparable.compareTo] on
+/// the objects. If any object is not [Comparable], this throws a [TypeError]
+/// (`CastError` on some SDK versions).
```

Comment:
We control the SDK version for Flutter, which error is it for us?

Reply:
That's an interesting question.  When I tried, this, it gave me:

```
The following _CastError was thrown building Home(dirty):
type 'Foo' is not a subtype of type 'Comparable<dynamic>' in type cast
```

The interesting thing is that `CastError` is deprecated in the SDK version we're using, with the reason "Use TypeError instead".

So, it seems that the SDK is throwing a private, deprecated error?

The definition of `_CastError` is:

```dart
class _CastError extends Error implements CastError, TypeError { /* ... */ }
```

So... I'm not sure what the right answer is.  I think I'd rather it be `TypeError`, but that's not what the user will see, at least for now.

Reply:
I'm not sure why the SDK's not sending a `_TypeError`, as that is defined similarly:

```dart
class _TypeError extends Error implements TypeError, CastError { /* ... */ }
```

Reply:
I guess we can leave the docs as-is then until its cleaned up in the SDK.

Reply:
If `CastError` implements `TypeError`, then it is a `TypeError` - the comment is not necessary

Reply:
Well, sure, but that's not what it prints when the error is shown.  It prints `_CastError`.

Reply:
Oh, right. A _CastError is a TypeError, so we should just mention the TypeError in the doc.

Reply:
But you can't use _CastError in your code anyways. I'd assume if you read the doc you are wondering what error type you're supposed to catch in a try-catch - and that would be a TypeError, no?

Reply:
It would.  That's why this is complicated.  Nowhere does the stack trace output say "TypeError", yet that is the type of the error that you need to catch.

Reply:
I think I'd like to leave it as just saying `TypeError`, and remove the parenthetical about `CastError` since that one is deprecated, and point out that the output may say `_CastError`.

Reply:
OK, I updated it to say that.  See what you think.